src/partner-admin/partner-admin-auth.guard.ts

```diff
@@ -8,26 +8,31 @@ export class PartnerAdminAuthGuard implements CanActivate {
   constructor(private authService: AuthService, private usersRepository: UserRepository) {}
 
   async canActivate(context: ExecutionContext): Promise<boolean> {
-    const request = context.switchToHttp().getRequest<Request>();
-
+    const request = await context.switchToHttp().getRequest<Request>();
```

Comment:
How come this now needs an await?

Reply:
Tested it and this was a mistake on my part haha. Good catch. I made the mock wrong. 👀  Lesson in that you've got to get those mocks right 😬

Reply:
Ahaha fair - mocking is quite tricky tbh!